nextcloud_connector.py

```diff
@@ -0,0 +1,395 @@
+
+    def _handle_delete_item(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        path = param['path'].lstrip('/').rstrip('/')
+        self.save_progress("path: {}".format(path))
+        result, rest_result = self._make_rest_call(action='delete', delete_path=path)
+        if result is False:
+            action_result.set_status(phantom.APP_ERROR, "Action failed")
+        summary = action_result.update_summary({})
+        summary['status'] = 'Success'
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_create_folder(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        path = param['path'].lstrip('/').rstrip('/')
+        self.save_progress("path: {}".format(path))
+        result, rest_result = self._make_rest_call(action='create_folder', create_folder=path)
+        if result is False:
+            action_result.set_status(phantom.APP_ERROR, "Action failed")
+        summary = action_result.update_summary({})
+        summary['status'] = 'Success'
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_download_folder(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        path = param['path'].lstrip('/').rstrip('/')
+        folder_name = path.split('/')[-1]
+        download_path = '/tmp/' + self._randomstr() + folder_name + '.zip'
+        self.save_progress("download_path: {}".format(download_path))
+        result, rest_result = self._make_rest_call(
+            action='download_folder', download_folder=path, download_path=download_path, file_name=folder_name + '.zip')
+        if result is False:
+            action_result.set_status(phantom.APP_ERROR, "Action failed")
+        summary = action_result.update_summary({})
+        summary['status'] = 'Success'
+        summary['vault_id'] = rest_result[2]
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_download_file(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        path = param['path'].lstrip('/').rstrip('/')
+        file_name = path.split('/')[-1]
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        download_path = '/tmp/' + self._randomstr() + file_name
+        self.save_progress("download_path: {}".format(download_path))
+        result, rest_result = self._make_rest_call(action='download_file', download_file=path, download_path=download_path, file_name=file_name)
+        if result is False:
+            action_result.set_status(phantom.APP_ERROR, "Action failed")
```

Comment:
Same, should probably check all of the actions.

Reply:
done